client/asset/interface.go

```diff
@@ -59,6 +61,18 @@ func (wt WalletTrait) IsRecoverer() bool {
 	return wt&WalletTraitRecoverer != 0
 }
 
+// IsWithdrawer tests if the WalletTrait has the WalletTraitSender bit set,
+// which indicates the presence of a Withdraw method.
+func (wt WalletTrait) IsWithdrawer() bool {
+	return wt&WalletTraitWithdrawer != 0
+}
+
+// IsSweeper test if the WalletTrait has the WalletTraitSweeper bit set, which
+// indicates the presence of a Sweep method.
+func (wt WalletTrait) IsSweeper() bool {
+	return wt&WalletTraitSweeper != 0
+}
```

Comment:
I still think the Sweeper interface can be removed as the purpose of making it was to deal with the issue that this pr solves. All `Withdrawer`s are also `Sweepers`.

Reply:
> I still think the Sweeper interface can be removed as the purpose of making it was to deal with the issue that this pr solves. All `Withdrawer`s are also `Sweepers`.


@chappjc, what do you think?

Reply:
`Sweeper` is an optional interface.  It does no harm to have it, and I'm pretty sure I would eventually add it back if it's not added now.

The method is more convenient than `Balance` check -> `Withdraw` -> check error in case balance was too low -> back to `Balance` check.

We're not talking about using it in the frontend in this PR are we?  I'm pretty sure the frontend demands a **UX** that is more in line with `Withdraw(specificAmount)`.  But I would absolutely use a Go `Sweep` method if I were writing automation logic to regularly send whatever is presently available to send (without the fuss of a logic race given active orders).

In many cases, it simplifies the backend handling if the API is (send whatever and tell me the amount sent) vs (spend exactly X and fail if I cannot, possibly leaving extra funds if more landed while I was studying the UI that displayed some available balance).  e.g. a `sweepaccount` RPC or special amount values like `"!"` for electrum can avoid the complex dexc application logic of picking utxos and dealing with the "subtract from amount" semantics required.

Reply:
Thanks for the explanation.

Reply:
> We're not talking about using it in the frontend in this PR are we? I'm pretty sure the frontend demands a UX that is more in line with Withdraw(specificAmount).

This can come in a separate PR? the `sweep` feature should fit the current UX/UI.

Reply:
>  the sweep feature should fit the current UX/UI.

Like a button on the send form? the current display might not be the total balance (assuming the user receives extra funds).

Reply:
Let's not worry about the frontend with respect to sweep right now.  Let's just address the send/withdraw issues you're after.
Sweeper can stay, but no need to implement or use any of it.  Maybe we'll get rid of it, but I don't think we should.

Maybe I'm in the minority, but a sweep feature accessible via a sweep RPC or a Sweep Go method is a very helpful thing to have even if that's not how we're designing our browser frontend.  It does no harm to leave it.